tripy/tripy/function_registry.py

```diff
@@ -197,14 +197,20 @@ def matches_type(name: str, annotation: type, arg: Any) -> bool:
         annotations = self._get_annotations()
 
         # Check if we have too many positional arguments. We can only do this if there isn't a variadic positional argument.
-        if not any(annotation.kind == inspect.Parameter.VAR_POSITIONAL for annotation in annotations.values()) and len(
-            args
-        ) > len(annotations):
+        # The variadic argument, if present, is necessarily the last.
+        annotation_items = list(annotations.items())
+        has_variadic_arg = (
+            (annotation_items[-1][1].kind == inspect.Parameter.VAR_POSITIONAL) if annotation_items else False
+        )
+        if not has_variadic_arg and len(args) > len(annotations):
             return Result.err(
                 [f"Function expects {len(annotations)} parameters, but {len(args)} arguments were provided."],
             )
 
-        for (name, annotation), arg in zip(annotations.items(), args):
+        positional_args_to_check = (
+            zip(annotation_items, args) if not has_variadic_arg else zip(annotation_items[:-1], args)
+        )
```

Comment:
What if we just repeat the annotation for variadic arguments here? Then we wouldn't need a separate loop or special handling

Reply:
It would check the first variadic argument only (that was the existing behavior, I tested it). We would still need separate logic to check the rest.

Reply:
Oh, I misunderstood the comment, you meant copying over the final annotation that many times. Yes, that would result in less code.